duration/main.go

```diff
@@ -0,0 +1,209 @@
+// Package duration provides a partial implementation of ISO8601 durations. (no months)
+package duration
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+var (
+	// ErrBadFormat is returned when parsing fails
+	ErrBadFormat = errors.New("bad format string")
+
+	ErrWeeksNotWithYearsOrMonth = errors.New("weeks are not allowed with years or months")
+
+	ErrMonthsInDurationUseOverload = errors.New("months are not allowed with the ToDuration method, use the overload instead")
+
+	full = regexp.MustCompile(`P((?P<year>\d+)Y)?((?P<month>\d+)M)?((?P<day>\d+)D)?(T((?P<hour>\d+)H)?((?P<minute>\d+)M)?((?P<second>\d+(?:\.\d+))S)?)?`)
+	week = regexp.MustCompile(`P((?P<week>\d+)W)`)
+)
+
+type Duration struct {
```

Comment:
we probably want the type to remain private to avoid anyone taking a dependency on it